libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -1412,10 +1412,10 @@ public void delete(String soupName, Long[] soupEntryIds, boolean handleTx) {
 	            db.beginTransaction();
 	        }
 	        try {
-	            db.delete(soupTableName, getSoupEntryIdsPredicate(soupEntryIds), (String []) null);
+	            db.delete(soupTableName, getSoupEntryIdsPredicate(soupEntryIds), new String[0]);
```

Comment:
delete no longer likes null for the third arg (tries to do a .length on it without checking for null).